### PR TITLE
[CI] docker images build script cmd line args optional

### PR DESCRIPTION
**Pull request by mherkazandjian**
i modified the ``./docker/build.sh`` script to allow building the docker images without the necessity of 
passing a command the build script. I also updated the documentation.
The main motivation for doing this change was that i found the readme a bit confusing since
it was not clear (for a newcomer like me) that ``build.sh`` builds and runs commands. I think it 
it is clearer now since it is clear in the readme that ``build.sh`` can be used to only build
the image and optionally run commands in the image that has been just built.

### CODE REVIEW

**Comment by mherkazandjian** [comment]
it seems there is a merge conflict, it is minor, i'll resolve it once i get feedback on this PR.

**Comment by tqchen** [comment]
@areusch can you help to manage this PR? Thank you

File: docker/README.md [hunk 1]
```diff
@@ -76,14 +77,39 @@ You can also start an interactive session by typing
 ./docker/build.sh image_name -it bash
 ```
 
+The built docker images are prefixed by ``tvm.``, for example the command
+
+````bash
+./docker/build.sh image_name 
+````
+
+produced the image ``tvm.ci_cpu`` that is displayed in the list of docker images
```
Comment (areusch) [inline docker/README.md:86]: minor nit: produces

File: docker/build.sh [hunk 2]
```diff
@@ -18,11 +18,11 @@
 # under the License.
 
 #
-# Execute command within a docker container
+# Build a docker container and optionally execute command within a it
 #
 # Usage: build.sh <CONTAINER_TYPE> [--tag <DOCKER_IMAGE_TAG>]
 #                [--dockerfile <DOCKERFILE_PATH>] [-it]
-#                [--net=host] [--cache-from <IMAGE_NAME>] <COMMAND>
+#                [--net=host] [--cache-from <IMAGE_NAME>] <COMMAND (optional)>
```
Comment (areusch) [inline docker/build.sh:25]: could you enclose command in `[]` to mark it optional, like the other options here? `[<COMMAND>]`

File: docker/build.sh [hunk 3]
```diff
@@ -164,25 +164,29 @@ if [[ $? != "0" ]]; then
...
-    ${COMMAND[@]}
+if [[ -n ${COMMAND} ]]; then
+
+    # Run the command inside the container.
+    echo "Running '${COMMAND[@]}' inside ${DOCKER_IMG_SPEC}..."
+
+    # By default we cleanup - remove the container once it finish running (--rm)
+    # and share the PID namespace (--pid=host) so the process inside does not have
+    # pid 1 and SIGKILL is propagated to the process inside (jenkins can kill it).
+    echo ${DOCKER_BINARY}
```
Comment (areusch) [inline docker/build.sh:175]: delete

**Review by areusch — COMMENTED** [review]
thanks @mherkazandjian , this looks pretty good and I agree requiring a command doesn't make sense for `docker/build.sh`. I have a couple minor comments on documentation and some cruft.

File: docker/README.md (hunk 1 above)
Comment (mherkazandjian) [reply docker/README.md]: done

File: docker/build.sh (hunk 2 above)
Comment (mherkazandjian) [reply docker/build.sh]: done

File: docker/build.sh (hunk 3 above)
Comment (mherkazandjian) [reply docker/build.sh:175]: Hi @areusch , i don't get what i should delete. The enclosing ``if`` ... ``fi`` or just the ``echo ${DOCKER_BINARY}``?

Comment (mherkazandjian) [reply docker/build.sh:175]: the comment snippet ``#By default ... echo ${DOCKER_BINARY}`` was already there from before. i this change i only wrapped it with the ``if ... fi``.

Comment (areusch) [reply docker/build.sh:175]: oh just the echo, I think it's debug, right?

Comment (areusch) [reply docker/build.sh:175]: oh, actually, this is fine to leave as-is, then.

**Review by areusch — APPROVED** [review]

**Comment by areusch** [comment]
thanks @mherkazandjian , the PR is now merged!

**Pull request merged**